### PR TITLE
tests for scripts

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@djbyrne mind have look and suggest what are the smallest env, to have these test running fast...

File: tests/models/rl/test_scripts.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_dqn(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.dqn_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:16]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env PongNoFrameskip-v4 --max_steps 10 --fast_dev True '
                                      '--warm_start_steps 10 --n_steps 4'])
```

File: tests/models/rl/test_scripts.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+from unittest import mock
+
+import pytest
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:6]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env PongNoFrameskip-v4 --max_steps 10 --fast_dev True '
                                      '--warm_start_steps 10 --n_steps 4'])
```

File: tests/models/rl/test_scripts.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_double_dqn(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.double_dqn_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:26]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env PongNoFrameskip-v4 --max_steps 10 --fast_dev True '
                                      '--warm_start_steps 10 --n_steps 4'])
```

File: tests/models/rl/test_scripts.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_dueling_dqn(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.dueling_dqn_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:36]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env PongNoFrameskip-v4 --max_steps 10 --fast_dev True '
                                      '--warm_start_steps 10 --n_steps 2 --batch_size 10'])
```

File: tests/models/rl/test_scripts.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_noisy_dqn(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.noisy_dqn_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:46]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env PongNoFrameskip-v4 --max_steps 10 --fast_dev True '
                                      '--warm_start_steps 10 --n_steps 2 --batch_size 10'])
```

File: tests/models/rl/test_scripts.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_per_dqn(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.per_dqn_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:56]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env CartPole-v0 --max_steps 10 --fast_dev True '
                                      '--n_steps 2 --batch_size 10'])
```

File: tests/models/rl/test_scripts.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+def test_cli_run_rl_reinforce(cli_args):
+    """Test running CLI for an example with default params."""
+    from pl_bolts.models.rl.reinforce_model import run_cli
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
+@pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3'])
```
Comment (djbyrne) [inline tests/models/rl/test_scripts.py:66]:
```suggestion
@pytest.mark.parametrize('cli_args', ['--env CartPole-v0 --max_steps 10 --fast_dev True '
                                      '--n_steps 2 --batch_size 10'])
```

File: tests/models/rl/test_scripts.py (hunk 2 above)
Comment (Borda) [reply tests/models/rl/test_scripts.py:6]: int it `fast_dev_run` ?

**Comment by Borda** [comment]
seems like `tests/models/self_supervised/test_models.py::test_amdim` is hanging...

**Comment by Borda** [comment]
any idea about this error:
```
    def weighted_mean(result, weights):
        weights = weights.to(result.device)
>       numerator = torch.dot(result.float(), weights.t().float())
E       RuntimeError: 1D tensors expected, got 4D, 1D tensors at ../aten/src/TH/generic/THTensorEvenMoreMath.cpp:83
```

**Comment by williamFalcon** [comment]
just tested and it's fine... looks like you need the latest PL

File: tests/models/test_executable_scripts.py [hunk 8]
```diff
@@ -1,30 +1,31 @@
 from unittest import mock
+
 import pytest
 
 
 @pytest.mark.parametrize('cli_args', ['--max_epochs 1 --limit_train_batches 3 --limit_val_batches 3 --batch_size 3'])
 def test_cli_basic_gan(cli_args):
-    from pl_bolts.models.gans.basic.basic_gan_module import cli_main
```
Comment (williamFalcon) [inline tests/models/test_executable_scripts.py:7]: please keep these as cli_main.... run_cli is not clear

File: pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py [hunk 9]
```diff
@@ -138,8 +138,7 @@ def add_model_specific_args(parent_parser):
         return parser
 
 
-# todo: covert to CLI func and add test
-if __name__ == '__main__':
+def cli_main():
```
Comment (williamFalcon) [inline pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py:141]: @Borda please keep these as cli_main...
run_cli is not clear

Comment (Borda) [reply pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py:141]: Ok, then let's rename all (before there were both)

**Comment by Borda** [comment]
> just tested and it's fine... looks like you need the latest PL

the tests are running `pytorch-lightning==0.9.0rc14`

**Comment by williamFalcon** [comment]
vae had a bug

**Comment by Borda** [comment]
it seems to be killed for some resources issue... @ananyahjha93 mind have look?

**Comment by Borda** [comment]
@ananyahjha93 can we merge this and in follow-up resolve the skipped tests?

**Pull request merged**